### ISSUE TITLE
Support aggregating multiple coverage reports

### ISSUE THREAD

**Issue by pbrisbin** [codeclimate/ruby-test-reporter#134]
This is a common feature-request and is currently represented as three independent Issues:
- #24 
- #35 
- #49 

I'd like to consolidate on a single Issue for tracking.

Use cases:
- Running multiple tests in parallel, each emitting part of the coverage
- Running different test suits (unit, integration), each emitting part of coverage
- Running multiple languages' suites in the same CI run

The partial coverages emitted by these separate processes should all be postable to codeclimate.com and visible (either aggregrated or separately) on the site.

The community has provided a workaround for some use cases: https://github.com/grosser/cc-amend Note: it might be possible to improve this work-around (or build a new, similar one) since the new usage style runs the reporter separately. I.e. you could inject a middle step to aggregate multiple payloads before invoking the reporter.

We definitely want to support this in a first-class way server side, but we don't yet have a timeline on the work,. We'll provide updates here as we have them.

**Comment by dblandin** [comment]
We did this!! 🎉 

https://github.com/codeclimate/test-reporter
https://codeclimate.com/changelog/591c99570d51f6481f001291

**Comment by streetlogics** [comment]
@dblandin Seems that when one is using two different test suites and parallelizing those tests that configuring things gets a bit tricky.  I played around with a few different circleci configs trying to get it working (using this as a starting point https://docs.codeclimate.com/docs/circle-ci-test-coverage-example), but it kept wanting to run every test instead of a subset of the tests for that specific container being parallelized.  Do you guys have an example for getting this working where the existing circle test section looks like this for parallelization and distributing tests evenly according to the time it takes to run them:

```
test:
  override:
    - mkdir -p $CIRCLE_TEST_REPORTS/cucumber:
        parallel: true
    -  PRECOMPILE=true bundle exec cucumber --format json --out $CIRCLE_TEST_REPORTS/cucumber/tests.cucumber:
        parallel: true
        files:
          - features/*.feature
    - bundle exec rspec --format doc --format RspecJunitFormatter --out $CIRCLE_TEST_REPORTS/rspec.xml:
        parallel: true
        files:
          - spec/**/*_spec.rb
```

**Comment by dblandin** [comment]
Hey @streetlogics,

Would you mind porting your comment over to https://github.com/codeclimate/test-reporter as a new issue? Thanks!
